compat/mingw.c

```diff
@@ -3337,6 +3338,45 @@ static int is_system32_path(const char *path)
 	return 1;
 }
 
+/*
+ * Extract the Windows OS build number from the registry, this is
+ * best method of those available here:
+ * https://stackoverflow.com/a/47583450/262458
+ *
+ * This post explains how to read the registry using WinAPI:
+ * https://stackoverflow.com/a/35717/262458
+ */
+unsigned windows_build(void)
+{
+	static unsigned is_cached    = 0;
+	static unsigned cached_value = 0;
+
+	if (is_cached)
+		return cached_value;
+
+	HKEY hkey;
```

Comment:
This is a declaration after a statement (the `if`), which is not allowed in Git's source code. Please move all the declarations to the beginning of the block.